sig/writer.go

```diff
@@ -0,0 +1,62 @@
+package sig
+
+import (
+	"crypto/cipher"
+	"github.com/dedis/crypto/random"
+	"hash"
+	"io"
+)
+
+// Create a Writer that interposes on underlying Writer wr,
+// hashing all bytes written and appending a trailing signature on Close.
+//
+// The signature is assumed to be fixed-size at least for a given public key,
+// and is appended to the end of the written data with no framing metadata.
+// Thus we can sign arbitrary-size streaming messages efficiently,
+// and the message's total size does not need to be known in advance.
+func Writer(wr io.Writer, key SecretKey, rand cipher.Stream) io.WriteCloser {
```

Comment:
I like the idea of automatically appending a signature on messages :) 
One small detail though: it seems more ... *natural* to pass a `Scheme` here instead of a `SecretKey`, as some signing mechanisms also need the public key for signing. Of course you can do that only having the `SecretKey` and calling `PublicKey()` on it, but the way I would like to read this method's signature is : "Give me a writer that appends signature  using this signing scheme", vs "Give me a writer that appends signature using this secret key".
Well, that's only semantics.. :)